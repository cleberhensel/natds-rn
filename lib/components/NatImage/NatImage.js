import React from 'react';
import PropTypes from 'prop-types';
import { View, Image } from './NatImage.styles';

export const NatImage = ({ source, onPress, type }) => (
  <View
    onPress={onPress}
    type={type}
  >
    <Image
      source={{ uri: source }}
      type={type}
    />
  </View>
);

NatImage.propTypes = {
  source: PropTypes.string,
  onPress: PropTypes.func,
  type: PropTypes.string,
};

NatImage.defaultProps = {
  type: 'default',
};
