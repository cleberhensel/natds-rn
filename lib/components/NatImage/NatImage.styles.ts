import styled from "styled-components/native";
const imageStyles = {
  card: {
    height: 184,
    width: "100%"
  },
  circle: {
    borderRadius: 100,
    height: 48,
    width: 48
  },
  default: {}
};
const getStylesFrom = styles => type => styles[type] || styles.default;
const getImageStyles = getStylesFrom(imageStyles);
export const View = styled.View`
  width: ${({ type }) => getImageStyles(type).width};
`;
export const Image = styled.Image`
  width: ${({ type }) => getImageStyles(type).width};
  height: ${({ type }) => getImageStyles(type).height};
  border-radius: ${({ type }) => getImageStyles(type).borderRadius || "0"};
`;
