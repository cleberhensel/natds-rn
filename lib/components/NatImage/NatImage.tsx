import React from "react";
import { View, Image } from "./NatImage.styles";
type NatImageProps = {
  onPress?: (...args: any[]) => any,
  source: string,
  type?: string
};
export const NatImage: React.SFC<NatImageProps> = ({
  source,
  onPress,
  type
}) => (
  <View onPress={onPress} type={type}>
    <Image source={{ uri: source }} type={type} resizeMode="contain" />
  </View>
);
NatImage.defaultProps = {
  onPress: () => {},
  type: "default"
};
