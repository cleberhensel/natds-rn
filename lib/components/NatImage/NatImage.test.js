import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';

import { NatImage } from './NatImage';

describe('NatImage component default', () => {
  it('Should render NatImage component ', () => {
    const button = renderer.create(
      <NatImage />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
