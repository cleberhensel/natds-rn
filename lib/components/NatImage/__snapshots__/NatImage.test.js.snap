// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatImage component default Should render NatImage component : NatImage snapshot 1`] = `
<View
  onPress={[Function]}
  style={
    Array [
      Object {
        "width": "",
      },
    ]
  }
  type="default"
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 0,
          "height": "",
          "width": "",
        },
      ]
    }
    type="default"
  />
</View>
`;
